Core/KSP.cs

```diff
@@ -92,9 +96,53 @@ private void Init()
                 foreach (DirectoryInfo subDirectory in directory.GetDirectories()) subDirectory.Delete(true);
             }
 
+            LoadCompatibleVersions();
+
             log.DebugFormat("Initialised {0}", CkanDir());
         }
 
+        public void SetCompatibleVersions(List<KspVersion> compatibleVersions)
+        {
+            this.compatibleVersions = compatibleVersions;
+            SaveCompatibleVersions();
+        }
+
+        private void SaveCompatibleVersions()
+        {
+            CompatibleKspVersionsDto compatibleKspVersionsDto = new CompatibleKspVersionsDto();
+
+            compatibleKspVersionsDto.versionOfKspWhenWritten = Version().ToString();
+            compatibleKspVersionsDto.compatibleKspVersions = compatibleVersions.Select(v => v.ToString()).ToList();
+
+            String json = JsonConvert.SerializeObject(compatibleKspVersionsDto);
+            File.WriteAllText(CompatibleKspVersionsFile(), json);
+
+            this.versionOfKspWhenCompatibleVersionsWereStored = Version();
+        }
+
+        private void LoadCompatibleVersions()
+        {
+            String path = CompatibleKspVersionsFile();
+            if (File.Exists(path))
+            {
+                string json = File.ReadAllText(path);
+                CompatibleKspVersionsDto compatibleKspVersionsDto = JsonConvert.DeserializeObject<CompatibleKspVersionsDto>(json);
+
+                compatibleVersions = compatibleKspVersionsDto.compatibleKspVersions.Select(v => KspVersion.Parse(v)).ToList();
+                this.versionOfKspWhenCompatibleVersionsWereStored = KspVersion.Parse(compatibleKspVersionsDto.versionOfKspWhenWritten);
+            }
+        }
+
+        private string CompatibleKspVersionsFile()
+        {
+            return Path.Combine(CkanDir(), "compatible_ksp_versions.json");
```

Comment:
As a sidenote we should really come up with a centralized mechanism to store both global and local settings data that is not the registry. But that's a story for another day.